### ISSUE TITLE
test_getTimezoneOffset is failing on FreeBSD

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2968]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2968|
|Type|defect|
|Created|2007-12-28 17:23:45Z|
|Branch|https://github.com/twisted/twisted/tree/test-timezone-2968|

The reason looks simple: the US/Eastern zone definition doesn't exist there. Using America/New_York seems to work, and doesn't break on Linux.

Attachments:

* [2968.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8f4/8f4b7a4f3e5c58eab322fc067ca3d00de50c0bc9/4fcb722687709c03036632c08ca48d80d1b6f5a1.diff) (541 bytes) - added by therve on 2007-12-29 22:14:02Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2968 2968
type__defect defect
reporter__therve therve
priority__highest highest
milestone__ 
branch__branches_test_timezone_2968 branches/test-timezone-2968
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1198862625000000 1198862625000000
changetime__1199653499000000 1199653499000000
version__None None
owner__ 
cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve removed owner|
|-|-|

Here is the simple fix.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @therve|
|-|-|

It might be better to switch to one of the other TZ formats.  Anything name-based is tied to platform libc details.

For example, "EST+5" (this is the same as US/Eastern or America/New_York, except it doesn't include daylight saving time adjustment, which I think is fine for this test).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

We do need daylight saving time adjustement for this test. I didn't find any other format that does the same thing :/. EST+5 doesn't work.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @therve|
|-|-|

There is a way to express DST with that format, but it's really complex and probably not worthwhile.  Switching to ```America/New_York``` is alright, I guess, as long as it works on all other platforms.  Did you try OS X and Windows?  If so and it works, please apply.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

(In [22284]) Branching to 'test-timezone-2968'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve commented|
|-|-|

(In [22285]) Change timezone id.

Refs [#2968](https://github.com/twisted/twisted/issues/2968)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set status to `closed`|
|-|-|

(In [22287]) Merge test-timezone-2968

Author: therve
Reviewer: exarkun
Fixes [#2968](https://github.com/twisted/twisted/issues/2968)

Replace the timezone used in tests so that the test passes under FreeBSD.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
